### PR TITLE
lib: do not filter out empty attrs in toLuaObject

### PR DESCRIPTION
**Pull request by horriblename**
there is a single use of `__empty` in the codebase, I need to double check what it's doing before I remove it

### CODE REVIEW

**Review by NotAShelf — DISMISSED** [review]
LGTM

**Comment by NotAShelf** [comment]
can be merged once @FrothyMarrow also approves

**Review by FrothyMarrow — COMMENTED** [review]
Add a warning for the users in case they are using `__empty`.

File: lib/lua.nix
Comment (NotAShelf) [inline lib/lua.nix:70]: Inherit `warn` from (lib.trivial) on top. We do not use aliases.

**Review by NotAShelf — CHANGES_REQUESTED** [review]
minor nit, LGTM otherwise

**Review by NotAShelf — APPROVED** [review]

**Pull request merged**
